OpenRA.Game/Network/LocalizedMessage.cs

```diff
@@ -123,13 +114,18 @@ public string Translate(ModData modData)
 			var argumentDictionary = new Dictionary<string, object>();
 			foreach (var argument in Arguments)
 			{
-				if (argument.Type == FluentArgument.FluentArgumentType.Number)
-					argumentDictionary.Add(argument.Key, new FluentNumber(argument.Value));
+				if (argument.Type is FluentArgument.FluentArgumentType.Number)
```

Comment:
I may be old-school, but the `is` here suggests a cast. `==` should be just fine.